packages/types/src/options.ts

```diff
@@ -144,4 +145,17 @@ export interface Options {
    * @returns The breadcrumb that will be added | null.
    */
   beforeBreadcrumb?(breadcrumb: Breadcrumb, hint?: BreadcrumbHint): Breadcrumb | null;
+
+  /**
+   * Function to compute tracing sample rate dynamically and filter unwanted traces.
+   *
+   * Can be defined in place of `tracesSampleRate`. If neither is defined, tracing is disabled.
+   *
+   * `startTransaction` or `startIdleTransaction` accept a `sampleContext` argument, which is then passed to this
```

Comment:
> It is a bit of a maintenance burden when we add this sort of cross documentation. I understand it makes sense now, but it is hard to keep consistent.

Switched to linking to the `startTransaction` docstring.

> Better to document SampleContext in the jsdoc for SampleContext.
>
> People looking at the API docs can look for where SampleContext appears -- the results would include startTransaction and any other place it may be used.

See https://github.com/getsentry/sentry-javascript/pull/2820#issuecomment-685826016.

> And we've been considering startIdleTransaction an internal thing for now, so it should not be prominent here.

Removed mention of it, as well as the ability to pass custom context to it at all. As you say, it's internal, so user-provided context doesn't make sense.